share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -160,6 +160,8 @@ default:
     # avoid moving targets like branches and tags
     SPACK_CONCRETIZER_REQUIRE_CHECKSUM: 1
     SPACK_BACKTRACE: 1
+    # Clone at a greater depth to allow generate early exit
+    GIT_DEPTH: 100
```

Comment:
On `gitlab.spack.io/spack/spack`, we are always testing a merge commit, doesn't that effectively make all changes at depth 1?

Reply:
hmm, yeah, I think you are right. Looking more closely at the discovered revisions it looks like it is grabbing the merge commit as the rev1 not the branch commit.

I am going to look at a few other methods to try and get this to work just using git, but it may be the case that fetching the PR diff from Github is going to be the best bet. It wouldn't be horrible to require prune untouched to require a touched list, but I would really like to be able to reliably deduce that without having to pass stuff in.